lib/dialects/abstract/query-generator.js

```diff
@@ -1642,12 +1654,19 @@ const QueryGenerator = {
             includeIgnoreAttributes: false
           }, topInclude.model);
         }
-        topLevelInfo.options.where['__' + throughAs] = this.sequelize.asIs([
+
+        const queryWhere = this.sequelize.asIs([
```

Comment:
Could some comments be added to explain why this change was needed for Oracle. I don't doubt it, it's just hard to understand at first glance.

Reply:
This part is not Oracle specific. It's a fix I've done for all versions but as I was working on Oracle on master branch, the fix has been done here.
The fix for V3 is here https://github.com/sequelize/sequelize/pull/7436 and explains the reason this code exists.

I know it's not a good idea to fix some bugs while doing a feature, but I had no time to make a special PR for master with this.